x/json_bool.go

```diff
@@ -0,0 +1,20 @@
+package x
+
+import (
+	"fmt"
+)
+
+// ConvertibleBoolean can unmarshal both booleans and strings.
+type ConvertibleBoolean bool
+
+func (bit *ConvertibleBoolean) UnmarshalJSON(data []byte) error {
+	asString := string(data)
+	if asString == "true" || asString == `"true"` {
+		*bit = true
+	} else if asString == "false" || asString == `"false"` {
+		*bit = false
+	} else {
+		return fmt.Errorf("boolean unmarshal error: invalid input %s", asString)
+	}
```

Comment:
```suggestion
	switch string(data) {
	case "true", `"true"`:
		*bit = true
	case "false", `"false"`:
		*bit = false
	default:
		return fmt.Errorf("boolean unmarshal error: invalid input %s", asString)
	}
```
I would probably write it like that, but that's just code style, so keep what you like more :wink:

Reply:
Thank you for your suggestion. I changed the code to use the `switch` statement and added some tests in https://github.com/ory/kratos/pull/1782/commits/70b1932b64d98d24afc4647ea8e988c8f58c195a